packages/openapi-generator/src/parser/operation.ts

```diff
@@ -64,102 +59,86 @@ export function getRelevantParameters(
-  pathPattern: string
-): OpenAPIV3.ParameterObject[] {
-  const pathParts = pathPattern.split('/');
-  const placeholders = pathParts.filter(part => isPlaceholder(part));
+  pathParameters: OpenApiParameter[]
+): string {
+  // Get the innermost curly bracket pairs with non-empty and legal content as placeholders.
+  const placeholders = pathPattern.match(/{[^/?#{}]+}/g);
+  // Get the non-parameter strings as static parts
+  const staticParts = pathPattern.split(/{[^/?#{}]+}/);
 
-  if (isValidPathPattern(pathPattern, placeholders)) {
-    throw new Error(
-      `Path pattern '${pathPattern}' is invalid or not supported.`
-    );
+  if (!placeholders) {
+    // No placeholder or path parameter found.
+    if (pathParameters.length > 0) {
+      throw new Error(
+        `Could not find placeholder for path parameter '${pathParameters
+          .map(pathParameter => pathParameter.originalName)
+          .join("', '")}'.`
+      );
+    }
+    return pathPattern;
   }
 
-  return placeholders.map(placeholder => {
+  const sortedPlaceholders = placeholders.map(placeholder => {
     const strippedPlaceholder = placeholder.slice(1, -1);
     const pathParameter = pathParameters.find(
-      param => param.name === strippedPlaceholder
+      param => param.originalName === strippedPlaceholder
     );
     if (!pathParameter) {
       throw new Error(
-        `Path parameter '${strippedPlaceholder}' provided in path is missing in path parameters.`
+        `Could not find path parameter for placeholder '{${strippedPlaceholder}}'.`
       );
     }
-
-    return pathParameter;
+    return `{${pathParameter.name}}`;
   });
-}
-/**
- * @internal
- */
-export function parsePathPattern(
-  pathPattern: string,
-  pathParameters: OpenApiParameter[]
-): string {
-  const pathParts = pathPattern.split('/');
-  const parameterNames = pathParameters.map(param => param.name);
 
-  return pathParts
-    .map(part => {
-      if (isPlaceholder(part)) {
-        if (!parameterNames.length) {
-          throw new Error(
-            `Could not find parameter for placeholder '${part}'.`
-          );
-        }
-        return `{${parameterNames.shift()}}`;
-      }
-      return part;
-    })
-    .join('/');
+  // Check if all path parameters match placeholders
+  for (const pathParameter of pathParameters) {
+    if (
+      !placeholders.find(
+        placeholder => placeholder.slice(1, -1) === pathParameter.originalName
+      )
+    ) {
+      throw new Error(
+        `Could not find placeholder for path parameter '${pathParameter.originalName}'.`
+      );
+    }
+  }
+
+  const newPathPattern: string[] = [];
+  while (staticParts.length > 0 && sortedPlaceholders.length > 0) {
+    newPathPattern.push(staticParts.shift()!, sortedPlaceholders.shift()!);
+  }
+  if (staticParts.length > 0) {
+    newPathPattern.concat(staticParts);
+  } else {
+    newPathPattern.concat(sortedPlaceholders);
+  }
+
+  return newPathPattern.join('');
```

Comment:
> 1. How do you know that the static part comes first?
I double-checked it. In case someone is also interested, according to the OpenAPI 3.0.3 documentation

> [...] The field name MUST begin with a forward slash (/). [...]

See more details on [https://swagger.io/specification/#patterned-fields](https://swagger.io/specification/#patterned-fields).

> 2. In JS you can just write `if (staticParts.length)`. Take a look at our [style guide](https://github.com/SAP/cloud-sdk-js/blob/main/STYLEGUIDE.md#use-truthyfalsy-checks-where-possible).

Thank you a lot for this. I was not aware of this style guide. It is very helpful!